staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/subresource.go

```diff
@@ -0,0 +1,192 @@
+	f := fieldpath.ManagedFields{}
+	t := map[string]*metav1.Time{}
+
+	// 1. Decode the scale entry and change it back to the parent API version and
+	// the parent path
+	scaleManagedFields, err := internal.DecodeObjectManagedFields(scaleEntries)
+	if err != nil {
+		return nil, err
+	}
+
+	for manager, versionedSet := range scaleManagedFields.Fields() {
+
+		if versionedSet.Set().Has(scalePath) {
+			// TODO (nodo): handle multiple managed fields versions.
+			newAPIVersion := h.parentGroupVersion.String()
+			newVersionedSet := fieldpath.NewVersionedSet(
+				fieldpath.NewSet(h.parentPath),
+				fieldpath.APIVersion(newAPIVersion),
+				versionedSet.Applied(),
+			)
+			f[manager] = newVersionedSet
+			if operationTime, ok := scaleManagedFields.Times()[manager]; ok {
+				t[manager] = operationTime
+			}
+		}
+	}
+
+	// If the subresource entry does not have the scale path, then do nothing.
+	if len(f) == 0 {
+		return h.parentEntries, nil
+	}
+
+	// 2. Decode the parent fields and process them to merge them with the
+	// subresource.
+	parentManagedFields, err := internal.DecodeObjectManagedFields(h.parentEntries)
+	if err != nil {
+		return nil, err
+	}
+
+	toRemove := fieldpath.NewSet(h.parentPath)
+	for manager, versionedSet := range parentManagedFields.Fields() {
+		// "steal" the scale path from the parent managed fields
+		newSet := versionedSet.Set().Difference(toRemove)
+		if newSet.Empty() {
+			continue
+		}
+
+		newVersionedSet := fieldpath.NewVersionedSet(
+			newSet,
+			versionedSet.APIVersion(),
+			versionedSet.Applied(),
+		)
+		f[manager] = newVersionedSet
+		if operationTime, ok := parentManagedFields.Times()[manager]; ok {
+			t[manager] = operationTime
+		}
+
+	}
+
+	return managedFieldsEntries(internal.NewManaged(f, t))
+}
+
+func matchScaleManager(fields map[string]fieldpath.VersionedSet, otherManagerID string) (string, fieldpath.VersionedSet, bool) {
+	// This matches if both scale and parent resources were using apply, or both using update
+	if existing, ok := fields[otherManagerID]; ok {
+		return otherManagerID, existing, true
+	}
+
+	// If scale and the parent used different operations we just check the name
+	// of the manager
+	var otherEntry metav1.ManagedFieldsEntry
+	if err := json.Unmarshal([]byte(otherManagerID), &otherEntry); err != nil {
+		return "", nil, false
+	}
+	manager := otherEntry.Manager
+
+	for managerJSON, set := range fields {
+		var entry metav1.ManagedFieldsEntry
+		if err := json.Unmarshal([]byte(managerJSON), &entry); err != nil {
+			return "", nil, false
+		}
+
+		if entry.Manager == manager {
+			return managerJSON, set, true
+		}
+	}
+	return "", nil, false
+}
+
+func managedFieldsEntries(entries internal.ManagedInterface) ([]metav1.ManagedFieldsEntry, error) {
+	obj := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	if err := internal.EncodeObjectManagedFields(obj, entries); err != nil {
```

Comment:
We should probably change that function signature if it requires us to do that weird dance here.